src/Common/LRUCache.h

```diff
@@ -117,6 +138,8 @@ class LRUCache
         auto token_it = insert_tokens.find(key);
         if (token_it != insert_tokens.end() && token_it->second.get() == token)
         {
+            // setImpl() may fail, but the final behavior seems not be affected
+            // next call of getOrSet() will still call load_func()
```

Comment:
I don't think this would resolve the problem.
1) before `set()`, we need to know whether the key is in cache or not, then calculate the extra size we need
2) in `getOrSet()`, we still cannot gurantee the value from `load_func()` would be inserted successfully becase we don't have the value size before `getOrSet()`

Reply:
> I prefer to make another lru cache class now.

Not now. I think may be after all your changes to LRUCache.cpp are ok (or a little changed version of it will be ok). I'll discuss with the team.

Reply:
Ok, now saw the comment below about it, so this is resolved.

Reply:
Ok, now saw the comment below about it, so this is resolved.